File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>


void getCipher(string input, int key){
    if(strlen(input)>0){
        for(int i = 0; i < strlen(input); i++){
            char letter = input[i];
            if(isalpha(letter)){
                if(isupper(letter)){
                    letter -= 65;
                    letter += key;
                    letter %= 26;
                    letter += 65;
                    printf("%c", letter);  
                }
                else if(islower(letter)){
                    letter -= 97;
                    letter += key;
                    letter %= 26;
                    letter += 97;
                    printf("%c", letter); 
                }
            }
            else
                printf("%c", letter); 
        }
    }
    printf("\n"); 
}


int main(int argc, string argv[]){
    if(argc == 2){
        if(atoi(argv[1]) > 0){
            printf("plaintext: ");
            string input = get_string();
            printf("ciphertext: ");
            getCipher(input,atoi(argv[1]));
            return 0;
        }
    }
    else{
        printf("Usage: ./caesar k\n");
        return 1;
    }
}

File: caesar.py
import sys

def main():
    if(len(sys.argv) == 2):
        plainInput = input("plaintext: ")
        key = int(sys.argv[1])
        print(getCipher(plainInput, key))
    else:
        print("Usage: python caesar.py k")


def getCipher(plainInput, key):
    if(len(plainInput) > 0):
        cipherOutput = ""
        for i in range(0, len(plainInput)):
            letter = plainInput[i]
            if(letter.isalpha()):
                    letter = ord(letter)
                    letter += key
                    cipherOutput += chr(letter)
            else:
                cipherOutput += letter
        return cipherOutput
    return ""
                    

if __name__ == "__main__":
    main()